controllers/storagecluster/storageconsumer.go

```diff
@@ -0,0 +1,177 @@
+			resources[subVolumeGroupNameKey] = "csi"
+			resources[csiRbdProvisionerSecretNameKey] = "rook-csi-rbd-provisioner"
+			resources[csiRbdNodeSecretNameKey] = "rook-csi-rbd-node"
+			resources[csiCephFsProvisionerSecretNameKey] = "rook-csi-cephfs-provisioner"
+			resources[csiCephFsNodeSecretNameKey] = "rook-csi-cephfs-node"
+			resources[rbdClientProfileNameKey] = "openshift-storage"
+			resources[cephFsClientProfileNameKey] = "openshift-storage"
+			resources[nfsClientProfileNameKey] = "openshift-storage"
+			resources[svgMetadataRadosNamespaceNameKey] = "csi"
+		} else {
+			resources[radosNamespaceNameKey] = localStorageConsumerName
+			resources[subVolumeGroupNameKey] = localStorageConsumerName
+			resources[csiRbdProvisionerSecretNameKey] = fmt.Sprintf("rbd-provisioner-%s", clientProfileName)
+			resources[csiRbdNodeSecretNameKey] = fmt.Sprintf("rbd-node-%s", clientProfileName)
+			resources[csiCephFsProvisionerSecretNameKey] = fmt.Sprintf("cephfs-provisioner-%s", clientProfileName)
+			resources[csiCephFsNodeSecretNameKey] = fmt.Sprintf("cephfs-node-%s", clientProfileName)
+			resources[rbdClientProfileNameKey] = clientProfileName
+			resources[cephFsClientProfileNameKey] = clientProfileName
+			resources[nfsClientProfileNameKey] = clientProfileName
+			resources[svgMetadataRadosNamespaceNameKey] = localStorageConsumerName
+		}
+		return nil
+	}); err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to create/update configmap %s: %v", configMap.Name, err)
+	}
+
+	storageConsumer.Name = localStorageConsumerName
+	storageConsumer.Namespace = storageCluster.Namespace
+	if _, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, storageConsumer, func() error {
+		if err := controllerutil.SetControllerReference(storageCluster, storageConsumer, r.Scheme); err != nil {
+			return err
+		}
+		// TODO: BackwardCompatibleAnnotation could be removed after convergence
+		util.AddAnnotation(storageConsumer, defaults.StorageConsumerBackwardCompatibleAnnotation, compatibleMode)
+		util.AddAnnotation(storageConsumer, defaults.StorageConsumerTypeAnnotation, defaults.StorageConsumerTypeLocal)
+
+		spec := &storageConsumer.Spec
+		spec.ResourceNameMappingConfigMap.Name = localStorageConsumerConfigMapName
+		spec.StorageClasses = []ocsv1a1.StorageClassSpec{
+			// TODO: after finding virt availability need to send corresponding sc
+			{Name: generateNameForCephBlockPoolSC(storageCluster)},
+			{Name: generateNameForCephFilesystemSC(storageCluster)},
+		}
+		spec.VolumeSnapshotClasses = []ocsv1a1.VolumeSnapshotClassSpec{
+			{Name: generateNameForSnapshotClass(storageCluster, rbdSnapshotter)},
+			{Name: generateNameForSnapshotClass(storageCluster, cephfsSnapshotter)},
+		}
+		spec.VolumeGroupSnapshotClasses = []ocsv1a1.VolumeGroupSnapshotClassSpec{
+			{Name: generateNameForGroupSnapshotClass(storageCluster, rbdGroupSnapshotter)},
+			{Name: generateNameForGroupSnapshotClass(storageCluster, cephfsGroupSnapshotter)},
+		}
```

Comment:
wont we have problem if the classes are disabled? how we are going to have that functionality?

Reply:
we obsoleted the functionality to disable classes in StorageCluster and by default above exists in all consumers so we don't have a functionality stopping default classes from distribution.